rosserial_arduino/src/ros_lib/ArduinoHardware.h

```diff
@@ -58,16 +58,17 @@
   // Stm32duino Maple mini USB Serial Port
   #define SERIAL_CLASS USBSerial
 #else 
-  #include <HardwareSerial.h>  // Arduino AVR
-  #if defined(USE_USBCDC)
-    // ESP32-S2 ESP32-C3
-    #define SERIAL_CLASS USBCDC
-  #elif defined(USE_HWCDC)
-    // ESP32-S3
-    #define SERIAL_CLASS HWCDC
-  #else
+
+  #if ARDUINO_USB_CDC_ON_BOOT 
+    #if ARDUINO_USB_MODE 
+      #define SERIAL_CLASS HWCDC
+    #else
+      #define SERIAL_CLASS USBCDC
+    #endif  
+  #else  
+    #include <HardwareSerial.h>  // Arduino AVR
     #define SERIAL_CLASS HardwareSerial
-  #endif
+  #endif 
```

Comment:
nit: trailing space added
```suggestion
  #endif
```